projects/js-packages/publicize-components/src/utils/use-post-meta.js

```diff
@@ -0,0 +1,77 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useCallback, useMemo } from '@wordpress/element';
+import { getShareMessageMaxLength } from './get-share-message-max-length';
+
+/**
+ * Returns the post meta values.
+ *
+ * @returns {import('./types').UsePostMeta} The post meta values.
+ */
+export function usePostMeta() {
+	const { editPost } = useDispatch( editorStore );
+
+	const metaValues = useSelect( select => {
+		const meta = select( editorStore ).getEditedPostAttribute( 'meta' ) || {};
+
+		const isPublicizeEnabled = meta.jetpack_publicize_feature_enabled ?? true;
+		const jetpackSocialOptions = meta.jetpack_social_options || {};
+		const attachedMedia = jetpackSocialOptions.attached_media || [];
+		const imageGeneratorSettings = jetpackSocialOptions.image_generator_settings ?? {
+			enabled: false,
+		};
+		const isPostAlreadyShared = meta.jetpack_social_post_already_shared ?? false;
+
+		const shareMessage = `${ meta.jetpack_publicize_message || '' }`.substring(
+			0,
+			getShareMessageMaxLength()
+		);
+		const shouldUploadAttachedMedia = jetpackSocialOptions.should_upload_attached_media ?? false;
+
+		return {
+			isPublicizeEnabled,
+			jetpackSocialOptions,
+			attachedMedia,
+			imageGeneratorSettings,
+			isPostAlreadyShared,
+			shareMessage,
+			shouldUploadAttachedMedia,
+		};
```

Comment:
Is this stable as it's a new object? I think it should be as we shouldn't get here without the meta changing, but if not, then this will invalidate the `useMemo` below

Reply:
Yes, it's stable because of the optimisation of `useSelect` via that dependency array.